### PR TITLE
✨ Add cancellation modal on schedule page

### PR DESCRIPTION
**Pull request by nicolasgti**
⚠️ _**This is a fork of the following branch:**_
- #83 

_**You should review and merge the above PR before this one.**_
## Description
Added modal on schedule page with action after pressing cancel button

## Preview
Captura de Tela 2022-02-04 às 17 25 26

### CODE REVIEW

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 1]
```diff
@@ -33,24 +36,18 @@ dayjs.extend(customParseFormat);
...
-      router.push({
-        pathname: "/team/projects/pac/schedule-status",
-        query: {
-          ...router.query,
-          error: errorCode,
-        },
-      });
-    }
-  }, [bookings, errorCode]); // eslint-disable-line
+  const [isActionDialogOpened, setIsActionDialogOpened] = useState<boolean>(true);
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:48]:
```suggestion
  const [isActionDialogOpened, setIsActionDialogOpened] = useState<boolean>(false);
```
I believe this was supposed to be false?

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 2]
```diff
@@ -84,10 +81,57 @@ export default function Schedules({ bookings, errorCode }: TSchedulesProps) {
     }
   };
 
+  const mutation = useMutation(cancelBooking, {
+    onSuccess: async (data) => {
+      console.info(data);
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:86]:
```suggestion
```
:fire:

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 3]
```diff
@@ -84,10 +81,57 @@ export default function Schedules({ bookings, errorCode }: TSchedulesProps) {
...
+      setIsCancelModalOpened(false);
+      setIsActionDialogOpened(false);
+      showToast("Agendamento cancelado com sucesso e removido de seus agendamentos.", "success");
+      if (selectedBooking) {
+        bookingRemove(selectedBooking);
+      }
+    },
+
+    onError: async (error) => {
+      console.info("errror", error);
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:96]:
```suggestion
```
:fire:

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 4]
```diff
@@ -84,10 +81,57 @@ export default function Schedules({ bookings, errorCode }: TSchedulesProps) {
...
+      </div>
+      <div className="flex flex-row justify-center">
+        <p className="text-gray-500 text-md">Sem Agendamentos</p>
+      </div>
+    </div>
+  );
+
+  if (bookings?.length != 0) {
+    EmptyState = <div></div>;
+  }
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:128]:
```suggestion
```
:fire:

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 5]
```diff
@@ -99,9 +143,18 @@ export default function Schedules({ bookings, errorCode }: TSchedulesProps) {
...
+            open={isCancelModalOpened}
+            handleClose={() => setIsCancelModalOpened(false)}
+            handleContinue={() => handleContinue()}
+          />
           <div className="flex-1 overflow-auto pb-4">
-            <p className="text-gray-500 text-sm mb-8">Para reagendar, selecione um agendamento.</p>
+            <p className="text-gray-500 text-sm mb-8">
+              Atendimentos podem ser reagendados com até <b>48h</b> de antecedência.
+            </p>
+            {EmptyState}
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:155]:
```suggestion
            {bookings?.length !== 0 && EmptyState}
```

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 6]
```diff
@@ -99,9 +143,18 @@ export default function Schedules({ bookings, errorCode }: TSchedulesProps) {
...
             {bookings &&
+              bookings.length >= 1 &&
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:157]:
```suggestion
```
:fire:

File: pages/team/projects/pac/schedule-status/schedules.tsx [hunk 7]
```diff
@@ -147,7 +202,7 @@ export default function Schedules({ bookings, errorCode }: TSchedulesProps) {
               })}
           </div>
         </div>
-        <Dialog open={!!selectedBooking}>
+        <Dialog open={!!selectedBooking && isActionDialogOpened}>
```
Comment (armr-dev) [inline pages/team/projects/pac/schedule-status/schedules.tsx:205]:
```suggestion
        <Dialog open={isActionDialogOpened}>
```
🔥

**Review by armr-dev — CHANGES_REQUESTED** [review]

**Review by uasouz — APPROVED** [review]

**Review by uasouz — APPROVED** [review]

**Review by armr-dev — APPROVED** [review]

**Pull request merged**